h2/src/main/org/h2/mvstore/ChunkParser.java

```diff
@@ -0,0 +1,270 @@
+/*
+ * Copyright 2004-2017 H2 Group. Multiple-Licensed under the MPL 2.0,
+ * and the EPL 1.0 (http://h2database.com/html/license.html).
+ * Initial Developer: H2 Group
+ */
+
+package org.h2.mvstore;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Namespace for methods used to parse a Chunk encoded as a String.
+ * A time efficient design intended to minimize the number of String objects
+ * created during parsing and minimize the number of String comparisons.
+ */
+public class ChunkParser {
+	public enum Field {
+		// TODO Using Java 8 replace this field -> key mapping code with method references.
+
+		// Serialized Chunk fields
+		// Each enum value contains the serialized key name, the value type, and
+		// a default value or a different field value to use as the default.
+		// Any new values here need code added to the setField() method below.
+		ID 				("chunk", Integer.class, 0), // ID requires a default value
+		BLOCK 			("block", Long.class, 0),
+		LEN 			("len", Integer.class, 0),
+		PAGE_COUNT 		("pages", Integer.class, 0),
+		PAGE_COUNT_LIVE ("livePages", Integer.class, PAGE_COUNT),
+		MAP_ID 			("map", Integer.class, 0),
+		MAX_LEN 		("max", Long.class, 0),
+		MAX_LEN_LIVE	("liveMax", Long.class, MAX_LEN),
+		META_ROOT_POS	("root", Long.class, 0),
+		TIME			("time", Long.class, 0),
+		UNUSED			("unused", Long.class, 0),
+		VERSION			("version", Long.class, ID),
+		NEXT			("next", Long.class, 0);
+
+		private final String key;
+    	private final Number defaultValue;
+    	private final Field keyForDefaultValue;
+    	private final Class<? extends Number> valueType;
+
+		private Field(String key, Class<? extends Number> valueType, Number defaultValue) {
+			this(key, valueType, defaultValue, null);
+		}
+		
+		private Field(String key, Class<? extends Number> valueType, Field keyForDefaultValue) {
+			this(key, valueType, null, keyForDefaultValue);
+		}
+		
+		private Field(String key, Class<? extends Number> valueType, Number defaultValue, Field keyForDefaultValue) {
+			this.key = key;
+			this.valueType = valueType;
+			this.defaultValue = defaultValue;
+			this.keyForDefaultValue = keyForDefaultValue;
+		}
+
+		public Number getDefaultValue() {
+			if (defaultValue == null) {
+				return keyForDefaultValue.getDefaultValue();
```

Comment:
this is not the same as the original logic - we need to look up the default value from the values in the parsed map